Add authentication and comment feature

gaurnett opened the pull request:
Hi Aditya and Taylor, 

I added an authentication feature that allows users to sign in and post comments. Auth allows me to now let specific users delete their own comments.

http://gaurnett-step-2020.uc.r.appspot.com/

taylorwan commented on portfolio/src/main/java/com/google/sps/data/User.java:
```diff
@@ -0,0 +1,24 @@
+            link = userService.createLoginURL("/");
+        }
+    }
+}
```
nit: missing newline

taylorwan commented on portfolio/src/main/webapp/comments.html:
```diff
@@ -1,114 +1,97 @@
-        <script src="./js/script.js"></script>
-    </body>
-</html>
+</html>
```
nit: newline

taylorwan commented on portfolio/src/main/webapp/css/style.css:
```diff
@@ -152,14 +152,28 @@ section {
+
+.hide-element {
+    display: none;
+}
```
nit: newline

taylorwan commented:
Also, it looks like my first comment now says "undefined" - can we default to the full name field when there is no user data?

gaurnett commented:
The first comment says undefined because I changed the name of the comment attribute from name to posterName. I deleted this comment from the datastore so this issue won't happen again. Also, this API does not give me access to fullname, only the username, so I have to continue asking for their name.
